scalableminds/webknossos
# Fix duplicating annotations via toggling show archived annotations button and fix downsample modal rendering errors

This PR fixes errors in the explorative annotation list view of the dashboard. It was possible to duplicate shown annotations by toggling the "show archived annotations" button. It was also possible to have the wrong actions rendered for an annotation.   
I additionally included a fix for the downsample modal not being rendered and even causing the layer settings to crash / not being rendered anymore.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
1.
- Go to the my annotations view in the dashboard
- make sure to have at least 2 annotations
- archive an annotation X
- click "show archived annotations"
- click "show unarchived annotations"
- annotation X should not appear again
- toggle more often between "show archived annotations" and show unarchived annotations". The should not be any duplicating of annotations.
2.
- Click on "show unarchived annotations" to show the archived annotations and then reload
- The shown annotations should be the exact same as before and should only include actually archived annotations.
- Now toggling between "show archived annotations" and "show unarchived annotations" should still not lead to an inconsistent state.
3.
- create volume annotation with only one magnification (for example, temporarily remove mags from a segmentation layer via the dataset settings; create volume annotation with fallback; read the mags to the dataset using the auto suggestions in the dataset settings view)
- click the orange "downsample" button in the left sidebar
- the downsample should open up happily 😄 

### Issues:
- fixes #6018 #6054

------
(Please delete unneeded items, merge only when none are left open)
